### PR TITLE
RA-1: Basic explorer functionality done and tested

### PR DESCRIPTION
**Pull request by dario-ramos**

### CODE REVIEW

File: bitprim.restapi/Controllers/AddressController.cs [hunk 1]
```diff
@@ -33,6 +33,10 @@ public ActionResult GetAddressHistory(string paymentAddress)
         {
             try
             {
+                if(!Validations.IsValidBase58Address(paymentAddress))
```
Comment (fpelliccioni) [inline bitprim.restapi/Controllers/AddressController.cs]: What about cashaddr?

File: bitprim.restapi/Controllers/ChainController.cs [hunk 2]
```diff
@@ -90,6 +90,19 @@ public ActionResult GetEstimateFee([FromQuery] int? nbBlocks = 2)
             }
         }
 
+        [HttpGet("/api/currency")]
```
Comment (fpelliccioni) [inline bitprim.restapi/Controllers/ChainController.cs]: Qué es esto? Esa constante es una cotización?

File: bitprim.restapi/Controllers/BlockController.cs [hunk 3]
```diff
@@ -162,6 +166,10 @@ public ActionResult GetBlocksByDate(int limit, string blockDate)
             {
                 return new Tuple<bool, string, DateTime?>(false, "Invalid limit; must be lower than " + config_.MaxBlockSummarySize, null);
             }
+            if(string.IsNullOrWhiteSpace(blockDate))
+            {
+                blockDate = DateTime.Now.Date.ToString(config_.DateInputFormat);
```
Comment (mdallago) [inline bitprim.restapi/Controllers/BlockController.cs]: you can use DateTime.Now.ToString(config_.DateInputFormat) or DateTime.Today.ToString(config_.DateInputFormat)

File: bitprim.restapi/Controllers/AddressController.cs (hunk 1 above)
Comment (dario-ramos) [reply bitprim.restapi/Controllers/AddressController.cs]: It is handled corrrectly by node-cint; will fix this method name

File: bitprim.restapi/Controllers/ChainController.cs (hunk 2 above)
Comment (dario-ramos) [reply bitprim.restapi/Controllers/ChainController.cs]: Yes, it's a valid currency response copied from Blockdozer. Will implement this feature in task RA-6

File: bitprim.restapi/Controllers/BlockController.cs (hunk 3 above)
Comment (dario-ramos) [reply bitprim.restapi/Controllers/BlockController.cs]: +1, will fix

**Pull request merged**